webviz_subsurface/plugins/_tornado_plotter_fmu/_plugin.py

```diff
@@ -0,0 +1,277 @@
+
+        # Reading from the csv file
+        if ensemble is not None and csvfile is not None:
+            ens_path = webviz_settings.shared_settings["scratch_ensembles"][ensemble]
+            self._parameter_provider = (
+                provider_factory.create_from_per_realization_parameter_file(ens_path)
+            )
+            self._table_provider = (
+                provider_factory.create_from_per_realization_csv_file(ens_path, csvfile)
+            )
+            self._ensemble_name = ensemble
+
+        elif aggregated_csvfile and aggregated_parameterfile is not None:
+            self._parameter_provider = provider_factory.create_from_ensemble_csv_file(
+                aggregated_parameterfile
+            )
+            self._table_provider = provider_factory.create_from_ensemble_csv_file(
+                aggregated_csvfile
+            )
+            try:
+                self._ensemble_name = self._table_provider.get_column_data(
+                    column_names=["ENSEMBLE"]
+                )
+            except KeyError:
+                self.error_message = f"'ENSEMBLE' is missing from {aggregated_csvfile}"
+
+        else:
+            self.error_message = (
+                "Specify either ensembles and csvfile or aggregated_csvfile "
+                "and aggregated_parameterfile"
+            )
+
+        try:
+            design_matrix_df = self._parameter_provider.get_column_data(
+                column_names=["SENSNAME", "SENSCASE"]
+            )
+        except KeyError:
+            self.error_message = (
+                "Required columns 'SENSNAME' and 'SENSCASE' is missing "
+                f"from {self._ensemble_name}. Cannot calculate tornado plots"
+            )
+
+        # Defining the datafame
+        design_matrix_df["ENSEMBLE"] = self._ensemble_name
+        design_matrix_df["SENSTYPE"] = design_matrix_df.apply(
+            lambda row: find_sens_type(row.SENSCASE), axis=1
+        )
+
+        self._tornado_widget = TornadoWidget(
+            webviz_settings=webviz_settings, realizations=design_matrix_df
+        )
+        self._responses: List[str] = self._table_provider.column_names()
+        if self._single_filters:
+            self._responses = [
+                response
+                for response in self._responses
+                if response not in self._single_filters
+            ]
+        if self._multi_filters:
+            self._responses = [
+                response
+                for response in self._responses
+                if response not in self._multi_filters
+            ]
+        self._initial_response: str = (
+            initial_response if initial_response else self._responses[0]
+        )
+
+        # Settingsgroup for switching between table and bars
+        self.add_store(
+            PlugInIDs.Stores.PlotPicker.BARS_OR_TABLE,
+            WebvizPluginABC.StorageType.SESSION,
+        )
+        self.add_shared_settings_group(
```

Comment:
Have you considered adding the settings group to the view instead of to the plugin as a shared settings group? If you want to add it to the view instead, you should move the file inside the new view folder (in the same manner as the view element; see other comment).

Reply:
Yes, I have done that on other plugins where the settingsdiffer between views, but in this plugin all settings are shared. Will it be an advatage of having the settings in the view here?